packages/react-core/src/components/Popover/PopoverHeader.tsx

```diff
@@ -1,25 +1,24 @@
 import * as React from 'react';
-import { Title, TitleSizes } from '../Title';
-import { PopoverContext } from './PopoverContext';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/Popover/popover';
+
+export interface PopoverHeaderProps extends React.HTMLProps<HTMLHeadingElement> {
+  /** Content of the popover header. */
+  children: React.ReactNode;
+  /** Indicates the header contains an icon. */
+  hasIcon?: boolean;
+  /** Class to be applied to the header. */
+  className?: string;
+}
 
 export const PopoverHeader: React.FunctionComponent<PopoverHeaderProps> = ({
   children,
-  id,
+  hasIcon = false,
+  className,
   ...props
 }: PopoverHeaderProps) => (
-  <PopoverContext.Consumer>
-    {({ headerComponent }) => (
-      <Title headingLevel={headerComponent} size={TitleSizes.md} id={id} {...props}>
-        {children}
-      </Title>
-    )}
-  </PopoverContext.Consumer>
+  <header className={css('pf-c-popover__header', className)} {...props}>
+    <h1 className={css(styles.popoverTitle, hasIcon && styles.modifiers.icon)}>{children}</h1>
```

Comment:
Should `headerComponent` set the heading level here, too.

Reply:
I'm not sure - I've been trying to figure out the real differences between `PopoverHeader` and `PopoverInternalHeader` and one of the main differences is that `PopoverInternalHeader` uses a `Title` component. @mcoker Do you think they should both use `Title` components?

Reply:
@kmcfaul is the only reason you built this to be composable because you needed to put the `hasIcon` prop at the outermost level?

Reply:
> @mcoker Do you think they should both use Title components?

@nicolethoen no, I think the popover title is better as an element of the popover component instead of the title component. That way we can change the styling (size, weight, etc) of the title in the CSS as opposed to needing a markup change, which is on the user to make in some cases. When we introduced the new Red Hat font, for example, the change in font face made either the modal or popover title (can't remember which, but both used the `<Title>` component) look different, so we ended up styling `.pf-c-title.pf-m-lg` (or whatever the class name was) to a different font-size in that component, which is a bad practice. Having the title be an element of a component won't put us in a situation like that if the title styles need to change.

Reply:
In our next upcoming breaking change, I'd like to get rid of the `<Title>` component in the popover and make the header and title elements introduced in this PR the default structural elements for the popover's title.

Reply:
@mcoker, I think you are right,  otherwise the heading level for the composable `PopoverHeader` is not configurable.